app/favorites/views.py

```diff
@@ -0,0 +1,114 @@
+@authentication_classes((SessionAuthentication, BasicAuthentication))
+@permission_classes((IsAuthenticated,))
+def splitMethod_favorite(request, favType, id, **kwargs):
+    if request.method == "DELETE":
+        return delete_favorite(request, favType, id)
+    elif request.method == "POST":
+        return create_favorite(request, favType, id)
+
+
+def create_favorite(request, favType, id, **kwargs):
+    types = dict((val.lower(), key) for (key, val) in Favorite.TYPE)
+    user = request.user
+    #obj_id = request.data['id']
+    obj_id = id
+    #favType = request.data['favType']
+    if favType not in types:
+        return Response(status=status.HTTP_400_BAD_REQUEST)
+    newFav = Favorite()
+    newFav.obj_id = obj_id
+    newFav.user = user
+    newFav.type = types[favType]
+    if newFav.save():
+        favSaveReturn = status.HTTP_201_CREATED
+    else:
+        favSaveReturn = status.HTTP_500_INTERNAL_SERVER_ERROR
+
+    return Response(favSaveReturn)
+
+def delete_favorite(request,  favType, id, **kwargs):
+    types = dict((val.lower(), key) for (key, val) in Favorite.TYPE)
+    user = request.user
+    obj_id = id
+    #obj_id = request.data['id']
+    #favType = request.data['favType']
+    if favType not in types:
+        return Response(status=status.HTTP_400_BAD_REQUEST)
+    if Favorite.objects.filter(user=user, type=types[favType], obj_id=obj_id).delete():
+        favSaveReturn = status.HTTP_200_OK
+    else:
+        favSaveReturn = status.HTTP_500_INTERNAL_SERVER_ERROR
+
+    return Response(favSaveReturn)
+
+@api_view(['GET'])
+@authentication_classes((SessionAuthentication, BasicAuthentication))
+@permission_classes((IsAuthenticated,))
+def get_favorites(request,  favType, **kwargs):
+    reqTypes = dict((val.lower(), key) for (key, val) in Favorite.TYPE)
+    user = request.user
+    #favType = request.data['favType']
+    if favType not in reqTypes:
+        return Response(status=status.HTTP_400_BAD_REQUEST)
+    favs = Favorite.objects.filter(user=user, type=reqTypes[favType])
+    jsonRetData = []
+    for fav in favs:
+        if reqTypes[favType] == "BOUN":
+            for obj in Bounty.objects.filter(standard_bounties_id=fav.obj_id):
+                ser = BountySerializer(obj)
+                jsonRetData.append(ser.data)
+        elif reqTypes[favType] == "KUDO":
+            for obj in Token.objects.filter(id=fav.obj_id):
+                ser = TokenSerializer(obj)
+                jsonRetData.append(ser.data)
+        elif reqTypes[favType] == "GRAN":
+            for obj in Grant.objects.filter(id=fav.obj_id):
+                ser = GrantSerializer(obj)
+                jsonRetData.append(ser.data)
+        elif reqTypes[favType] == "USER":
+            for obj in Profile.objects.filter(user=fav.obj_id):
+                ser = ProfileSerializer(obj)
+                jsonRetData.append(ser.data)
+
+    return Response(json.dumps(jsonRetData, indent=4, sort_keys=True, default=str))
```

Comment:
The responses are not json formatted 
![image](https://user-images.githubusercontent.com/1358093/63058978-8d991c80-bec4-11e9-903f-22bfc098d5aa.png)

Reply:
return Response(**json.loads**(json.dumps(jsonRetData, indent=4, sort_keys=True, default=str)))
can solve it

Reply:
My bad. I had to play around with recursion and avoiding errors that I probabbly forgot about that. I'll add your fix in